### DISCUSSION TITLE
allow to delete sessions

### DISCUSSION THREAD

**Discussion by morganveyret** [langfuse/langfuse#1428] [Ideas]
### Describe the feature or potential improvement

Would be nice to allow removal of session or at least remove sessions when they are empy (i.e. no traces linked to the session).
Maybe I'm not using the tool like I should but sometimes a session is just "failed" and has not real value that justifies keeping it.

**Comment by marcklingen** [comment]
Thanks for sharing, do you manually delete traces which results in empty sessions or why are there sessions without traces? I do not understand the 'failing' part

**Reply by morganveyret** [reply]
To clarify a bit more: I'm using langfuse for tracing in development mode, not production. So sometimes a session can be considered as "failed" (e.g. some bug in the code so results are irrelevant)

**Comment by morganveyret** [comment]
You're right, it goes like this:
- I run my code with a specific session id
- For some reason it fails or I'm not satisfied with this session
- I delete all session traces to try to keep a clean set of traces
- There are now empty sessions

**Reply by marcklingen** [reply]
Got it, that makes sense. What do you use a session for while in local dev?

**Reply by morganveyret** [reply]
At the moment I'm exploring agent development and I have like a test script which for example asks a series of questions to the agent.
So each time I run this script I do it in a session so I can have a nice overview of inputs/outputs at a high level and grouping of related traces (but maybe I got sessions wrong ).
Another case is like comparing rag and agent for question answering, each one get its own session.
(Reminds me another nice thing to have would be showing metrics per sessions or groups of sessions like you can already do with the user)

**Reply by marcklingen** [reply]
thanks for sharing, we are looking to expand the datasets feature which should cover most of this then: https://langfuse.com/docs/datasets/overview

**Reply by morganveyret** [reply]
Nice, didn't check the datasets  but I will try it !

**Comment by erantr1** [comment]
@marcklingen 
As far as I see sessions still can't be deleted.
Is it planned to allow such deletions?
My use-case: I just created some sessions to test this feature in my code, and I don't want them to appear anymore.
In general, some sessions might get irrelevant at some point, and instead of being deleted they will pile up as "trash".

**Comment by hemisalo** [comment]
Not being able to delete sessions seems almost like a bug to me. "Empty" sessions don't show up in the Sessions Observability UI, but the /api/public/sessions​ api endpoint returns them and there's no way to filter them out. 

Yes metrics api exists but even in the low thousands of traces, iterating them and tallying up the sessionIds is inefficient. 

This was rather confusing on our analytics dashboard reported to us by an end user as a bug because their analytics said there's more conversations reported than actual conversations they could analyze.
